test/e2e/declarative_test_fwk/helper.py

```diff
@@ -110,21 +110,36 @@ def delete(self, reference: k8s.CustomResourceReference) -> None:
         sleep(self.DEFAULT_WAIT_SECS)
         self.wait_for_delete(reference)  # throws exception if wait fails
 
-    def assert_expectations(self, verb: str, input_data: dict, expectations: model.ExpectDict, reference: k8s.CustomResourceReference) -> None:
+    def assert_expectations_k8s(self, verb: str, input_data: dict, expectations_k8s: model.ExpectDictK8S, reference: k8s.CustomResourceReference) -> None:
         """Asserts custom resource reference inside Kubernetes cluster against the supplied expectations
 
-        :param verb: expectations after performing the verb (apply, patch, delete)
+        :param verb: expectations_k8s after performing the verb (apply, patch, delete)
         :param input_data: input data to verb
-        :param expectations: expectations to assert
+        :param expectations_k8s: expectations to assert
         :param reference: custom resource reference
         :return: None
         """
-        self._assert_conditions(expectations, reference, wait=False)
-        # conditions expectations met, now check current resource against expectations
+        self._assert_conditions(expectations_k8s, reference, wait=False)
+        # conditions expectations met, now check current resource against expectations_k8s
         resource = k8s.get_resource(reference)
-        self.assert_items(expectations.get("status"), resource.get("status"))
+        # status and spec assertions with resources from Kubernetes
+        self.assert_items_k8s(expectations_k8s.get("status"), resource.get("status"))
+        self.assert_items_k8s(expectations_k8s.get("spec"), resource.get("spec"))
 
-        # self._assert_state(expectations.get("spec"), resource)  # uncomment to support spec assertions
+    def assert_expectations_boto3(self, verb: str, input_data: dict, expectations_boto3: model.ExpectDictBOTO3, reference: k8s.CustomResourceReference) -> None:
+        """Asserts custom resource reference in AWS MemoryDB service against the supplied expectations
+
+        :param verb: expectations_boto3 after performing the verb (apply, patch, delete)
+        :param input_data: input data to verb
+        :param expectations_boto3: expectations to assert
+        :param reference: custom resource reference
+        :return: None
+        """
+        resource = k8s.get_resource(reference)
+        # assertions with resources from boto3
+        # spec of k8s resource here is for getting name of resource
+        # boto3 call APIs using name of resource
+        self.assert_items_boto3(expectations_boto3, resource.get("spec"))
```

Comment:
This line implies that we can only assert that the spec fields of our resource match the response given by boto3. There are times where we may want to check that the status fields, too. For example, I know we have custom code for adding `events` into the status - we should check for that here too? Make sure that the # of events in the status matches the # of events returned by boto3

Reply:
`resource.get("spec")` here is only used for getting `name` of resources. Then we call APIs from boto3 by using `name` of resources. API returns both `spec` and `status`. Hence, it already covers `status`